### PR TITLE
Index rust

### PR DESCRIPTION
**Pull request by nagato-chan**

### CODE REVIEW

File: lib/src/main/java/space/taran/arklib/index/RustResourcesIndex.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package space.taran.arklib.index
+
+import java.nio.file.Path
+import java.nio.file.attribute.FileTime
+typealias ResourceId = Long
```
Comment (kirillt) [inline lib/src/main/java/space/taran/arklib/index/RustResourcesIndex.kt:5]: We need to use the same `ResourceId` as in `arklib`, i.e. CRC-32 hash and also filesize.

File: lib/src/main/java/space/taran/arklib/index/RustResourcesIndex.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+typealias ResourceMetas = MutableMap<Path, ResourceMeta>
```
Comment (kirillt) [inline lib/src/main/java/space/taran/arklib/index/RustResourcesIndex.kt:7]: No need for mutable structure here.

**Review by mdrlzy — COMMENTED** [review]
Kotlin tests looks fine
I think we can extract plain and aggregated resource index too.

File: arklib/src/signature.rs [hunk 3]
```diff
@@ -2,4 +2,6 @@
 
 pub const LONG: &'static str = "Ljava/lang/Long;";
 
+pub const INTERGER: &'static str = "Ljava/lang/Interger";
```
Comment (kirillt) [inline arklib/src/signature.rs:5]: "Integer", not "interger" (same type in multiple files).

**Comment by kirillt** [comment]
This PR can contain useful code, but it is too complicated and doesn't seem to solve the task at this moment.
We should approach the problem another way.

**Pull request closed without merging**